File: model/src/main/java/no/nav/pensjon/vtp/testmodell/repo/JournalRepository.java
package no.nav.pensjon.vtp.testmodell.repo;

import java.util.List;
import java.util.Optional;

import no.nav.pensjon.vtp.testmodell.dokument.modell.DokumentModell;
import no.nav.pensjon.vtp.testmodell.dokument.modell.JournalpostModell;

public interface JournalRepository {
    Optional<DokumentModell> finnDokumentMedDokumentId(String dokumentId);

    List<JournalpostModell> finnJournalposterMedFnr(String fnr);

    List<JournalpostModell> finnJournalposterMedSakId(String sakId);

    Optional<JournalpostModell> finnJournalpostMedJournalpostId(String journalpostId);

    String leggTilJournalpost(JournalpostModell journalpostModell);
}

File: model/src/main/java/no/nav/pensjon/vtp/testmodell/repo/JournalRepository.kt
package no.nav.pensjon.vtp.testmodell.repo

import no.nav.pensjon.vtp.testmodell.dokument.modell.DokumentModell
import no.nav.pensjon.vtp.testmodell.dokument.modell.JournalpostModell

interface JournalRepository {
    fun finnDokumentMedDokumentId(dokumentId: String): DokumentModell?
    fun finnJournalposterMedFnr(fnr: String): List<JournalpostModell>
    fun finnJournalposterMedSakId(sakId: String): List<JournalpostModell>
    fun finnJournalpostMedJournalpostId(journalpostId: String): JournalpostModell?
    fun save(journalpostModell: JournalpostModell): JournalpostModell
}
